### PR TITLE
Test static route

### PR DESCRIPTION
**Pull request by code-asher**
Mostly just wanted to add a test for static auth.

### CODE REVIEW

File: test/unit/helpers.test.ts
Comment (jsjoeio) [inline test/unit/helpers.test.ts:6]: I like how you separated this and added a comment 👏

File: test/unit/routes/static.test.ts
Comment (jsjoeio) [inline test/unit/routes/static.test.ts:36]: question: a bit confused here. I know the test is testing the `/static` route when unuathed but why do we expect a `404` when you fetch the route and then a `200` when fetching a file?

And why the `/-file` instead of `/file for `/static/-${testFile}?

File: test/unit/routes/static.test.ts
Comment (jsjoeio) [inline test/unit/routes/static.test.ts:38]: nit: testname should explain what the test is testing for -> 
```suggestion
  it("should return a 401 if authentication is enabled and user requests files", async () => {
```

File: test/unit/routes/static.test.ts
Comment (jsjoeio) [inline test/unit/routes/static.test.ts:26]: nit: test should explain what the test is testing for ->
```suggestion
  it("should return a 200 if not authed and requesting files", async () => {
```

File: test/unit/routes/static.test.ts
Comment (jsjoeio) [inline test/unit/routes/static.test.ts:7]:
```suggestion
describe("/static", () => {
```

**Review by jsjoeio — CHANGES_REQUESTED** [review]
Looks great! 🚀 

Mostly nits around test names and then a question

File: test/unit/routes/static.test.ts
Comment (code-asher) [reply test/unit/routes/static.test.ts:36]: The static endpoint expects a commit (for cache purposes, in this case I'm just using `-`) and then the full path of the file to serve.

So anything other than that will result in a 404 (since no path means no file). We could also do `/static/-`, `/static/-/some/path/that/probably/doesnot/exist`, etc.

Comment (jsjoeio) [reply test/unit/routes/static.test.ts:36]: Oh TIL. I didn't know how it worked so thanks for explaining that!

> We could also do...

Your call!

**Comment by code-asher** [comment]
I need to refactor the logger module mock because it causes errors in the e2e tests but that conflicts with some constant test changes you're making so I'm going to wait for that to get merged then rebase.

**Comment by jsjoeio** [comment]
> I'm going to wait for that to get merged then rebase.

**Review by jsjoeio — APPROVED** [review]
Nice work!

**Pull request merged**
